apis/pkg/meta/v1alpha1/provider_types.go

```diff
@@ -51,3 +60,75 @@ type Provider struct {
 
 	Spec ProviderSpec `json:"spec"`
 }
+
```

Comment:
The below conversion functions are new. They're identical to the equivalent v1 conversion functions.